contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,751 @@
+            dependencyType.scriptCount > 0,
+            dependencyType.scriptCount
+        );
+    }
+
+    /**
+     * @notice Overrides the script type and version that 
+     * would be returned by the core contract (`_contractAddress`)
+     * for a given project  (`projectId`) with the given dependency
+     * type (`_dependencyTypeId`).
+     * @param _contractAddress Core contract address.
+     * @param _projectId Project to override script type and version for.
+     * @param _dependencyTypeId Dependency type to return for project.
+     */
+    function addProjectDependencyOverride(
+        address _contractAddress,
+        uint256 _projectId,
+        bytes32 _dependencyTypeId
+    ) external onlyAdminACL(this.addProjectDependencyOverride.selector) {
+        require(
+            _dependencyTypes.contains(_dependencyTypeId),
+            "Dependency type is not registered"
+        );
+        projectDependencyOverrides[_contractAddress][
+            _projectId
+        ] = _dependencyTypeId;
+
+        emit ProjectDependencyOverrideAdded(
+            _contractAddress,
+            _projectId,
+            _dependencyTypeId
+        );
+    }
+
+    /**
+     * @notice Removes the script type and version override for a given 
+     * project (`projectId`) on a given core contract (`_contractAddress`).
+     * @param _contractAddress Core contract address.
+     * @param _projectId Project to remove override for.
+     */
+    function removeProjectDependencyOverride(
+        address _contractAddress,
+        uint256 _projectId
+    ) external onlyAdminACL(this.addProjectDependencyOverride.selector) {
+        delete projectDependencyOverrides[_contractAddress][_projectId];
+
+        emit ProjectDependencyOverrideRemoved(
+            _contractAddress,
+            _projectId
+        );
+    }
+
+    /**
+     * @notice Returns the dependency type for a given project (`projectId`) 
+     * on a given core contract (`_contractAddress`). If no override is set,
+     * the core contract is called to retrieve the script type and version as
+     * dependency type.
+     * @param _contractAddress Core contract address.
+     * @param _projectId Project to return dependency type for.
+     * @return dependencyTypeId Dependency type used by project.
+     */
+    function getDependencyForProject(
+        address _contractAddress,
+        uint256 _projectId
+    ) external view returns (string memory) {
+        bytes32 dependencyType = projectDependencyOverrides[_contractAddress][
+            _projectId
+        ];
+        if (dependencyType != bytes32(0)) {
+            return dependencyType.toString();
+        }
+
+        try
+            IGenArtDependencyConsumer(_contractAddress).projectScriptDetails(
+                _projectId
+            )
+        returns (string memory scriptTypeAndVersion, string memory, uint256) {
+            return scriptTypeAndVersion;
+        } catch {
+            return "";
```

Comment:
should we potentially revert when encountering this case?

Reply:
Yeah I think reverting with an unknown-contract-project pairing type message here makes sense.

Also, having a specific admin-defined set of supported contracts would be useful from the perspective of being able to throw a more idiomatic error message in this method in the case where contract is unknown vs. where e.g. an override is not specified vs. where contract is not "new enough" (not V3+).